plugins/inputs/kafka_consumer/kafka_consumer.go

```diff
@@ -109,7 +111,12 @@ func (k *KafkaConsumer) Init() error {
 	cfg := sarama.NewConfig()
 
 	// Kafka version 0.10.2.0 is required for consumer groups.
-	cfg.Version = sarama.V0_10_2_0
+	// Try to parse version from config. If can not, set default
+	if KafkaVersion, err := sarama.ParseKafkaVersion(k.Version); err == nil {
+		cfg.Version = KafkaVersion
+	} else {
+		cfg.Version = sarama.V0_10_2_0
+	}
```

Comment:
How about
```suggestion
	// Try to parse version from config. If can not, set default
	cfg.Version = sarama.V0_10_2_0
	if k.Version != "" {
		version, err := sarama.ParseKafkaVersion(k.Version)
		if err != nil {
			return fmt.Errorf("invalid version: %w", err)
		}
		cfg.Version = version
	}
```

as this will be strict with the user input. In your version we silently fallback to `0.10.2.0` if you use a malformed version.